### DISCUSSION TITLE
如何截获Connector连接失败的错误处理？

### DISCUSSION THREAD

**Discussion by yorkxu** [chenshuo/muduo#584] [General]
看了下代码，在使用TcpClient连接服务器时，如果使用io线程去处理connect过程，在连接失败的情况下，要么重试，要么报错，线程退出。如何才能截获连接失败错误处理？

**Comment by lys861205** [comment]
我感觉在做客户端的时候，同步方式是比较好的，muduo不太适合用作client

**Comment by fortunely** [comment]
Connector::handleError里面有简单的错误处理，以及retry尝试。如果想要自行处理错误，我的想法是可以在这里添加一个回调errorCallback_，然后，让用户层可以通过TcpClient来注册。
